File: system_simulations/network_model/cpp/discrete_event/Time.cpp
#ifndef TIME
#define TIME

class Time {
	public:
		Time(double real, int discrete){
			this->real = real;
			this->discrete = discrete;
		}

		Time* advance(Time* interval){
			if (interval->real > 0){
				return new Time(real + interval->real, 0);
			} else {
				return new Time(real, discrete + interval->discrete);
			}
		}

		double get_real(){
			return real;
		}

		int get_discrete(){
			return discrete;
		}

	private:
		double real = 0.0;
		int discrete = 0;
};

#endif

File: system_simulations/basketball_simulation/Time.java
import java.math.BigDecimal;

public class Time {
	private BigDecimal real;
	private int discrete;

	public Time(BigDecimal real, int discrete){
		this.real = real;
		this.discrete = discrete;
	}

	public Time advance(Time interval){
		if (!(interval.real.equals(0.0))){
			return new Time(real.add(interval.real), 0); // go to the right
		} else {
			return new Time(real, discrete + interval.discrete); // go up
		}
	}

	public BigDecimal getReal(){
		return real;
	}

	public int getDiscrete(){
		return discrete;
	}

	public String toString(){
		return real.toString();
	}
}
